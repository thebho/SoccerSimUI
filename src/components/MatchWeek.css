.MatchWeek {
  padding: 30px;
}
