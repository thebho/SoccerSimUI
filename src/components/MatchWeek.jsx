// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import type { Match, Season } from '../model';
import Matches from './Matches';
import './MatchWeek.css';
import SSButton from './SSButton';
import { simWeek } from '../actions/matches';
import { advanceWeek } from '../actions/season';

type Props = {
  matches?: Array<Match>,
  season: Season,
  simWeek: Function,
  advanceWeek: Function,
  weekComplete: boolean,
};

export class MatchWeek extends Component <Props> {
  simWeek = () => {
    this.props.simWeek(this.props.season.matchWeek, this.props.season.name );
  };

  advanceWeek = () => {
    if (this.props.season.matchWeek < 38) {
      this.props.advanceWeek(this.props.season.matchWeek+1, this.props.season.name);
    } else {
      console.error('Advancing season not supported');
    };
  }

  render() {
    if (this.props.season == null) {
      return null;
    };
    const { matches, season } = this.props;
    return (
      <div className='MatchWeek'>
        <h2>Week {season.matchWeek}</h2>
        <p>{season.name}</p>
        <Matches weeklyMatches={matches}/>
        <SSButton label='Sim Week' onClick={this.simWeek} disabled={this.props.weekComplete}/>
        <SSButton label='Advance Week' onClick={this.advanceWeek} disabled={!this.props.weekComplete}/>
      </div>
    );
  };
};


const mapStateToProps = (state: Object): Object => {
  return {
    matches: state.matchesReducer.matches,
    season: state.seasonReducer.season,
    weekComplete: state.seasonReducer.weekComplete,
  };
};
const mapDispatchToProps = (dispatch: Function): Object => {
  return {
    simWeek: (week: number, season: string) => dispatch(simWeek(week,season)),
    advanceWeek: (week: number, season: string ) => dispatch(advanceWeek(week, season)),
  };
};
â€Š
export default connect(mapStateToProps, mapDispatchToProps)(MatchWeek);
