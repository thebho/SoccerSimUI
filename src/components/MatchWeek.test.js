import React from 'react';
import ReactDOM from 'react-dom';
import { MatchWeek } from './MatchWeek';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const mockMatches = [
  {
    ID: 1,
    HomeTeam: 'ARS',
    AwayTeam: 'TOT',
    HomeTeamGoals: null,
    AwayTeamGoals: null,
    MatchWeek: 1,
    Season: 'TestSeason',
  },
  {
    ID: 2,
    HomeTeam: 'WHA',
    AwayTeam: 'CHE',
    HomeTeamGoals: null,
    AwayTeamGoals: null,
    MatchWeek: 1,
    Season: 'TestSeason',
  },
];
describe('MatchWeek', () => {
  it('matches snapshot with matches', () => {
    const wrapper = shallow(
      <MuiThemeProvider>
        <MatchWeek
          season={{ name: 'TestSeason', matchWeek: 1 }}
          matches={mockMatches}
        />
      </MuiThemeProvider>
    ).dive();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders without matches', () => {
    const wrapper = shallow(
      <MuiThemeProvider>
        <MatchWeek season={{ name: 'TestSeason', matchWeek: 1 }} />
      </MuiThemeProvider>
    ).dive();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
