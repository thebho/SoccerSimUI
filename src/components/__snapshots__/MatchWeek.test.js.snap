// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchWeek matches snapshot with matches 1`] = `
<div
  className="MatchWeek"
>
  <h2>
    Week 
    1
  </h2>
  <p>
    TestSeason
  </p>
  <Matches
    weeklyMatches={
      Array [
        Object {
          "AwayTeam": "TOT",
          "AwayTeamGoals": null,
          "HomeTeam": "ARS",
          "HomeTeamGoals": null,
          "ID": 1,
          "MatchWeek": 1,
          "Season": "TestSeason",
        },
        Object {
          "AwayTeam": "CHE",
          "AwayTeamGoals": null,
          "HomeTeam": "WHA",
          "HomeTeamGoals": null,
          "ID": 2,
          "MatchWeek": 1,
          "Season": "TestSeason",
        },
      ]
    }
  />
  <SSButton
    disabled={false}
    label="Sim Week"
    onClick={[Function]}
  />
  <SSButton
    disabled={true}
    label="Advance Week"
    onClick={[Function]}
  />
</div>
`;

exports[`MatchWeek renders with weekComplete 1`] = `
<div
  className="MatchWeek"
>
  <h2>
    Week 
    1
  </h2>
  <p>
    TestSeason
  </p>
  <Matches />
  <SSButton
    disabled={true}
    label="Sim Week"
    onClick={[Function]}
  />
  <SSButton
    disabled={false}
    label="Advance Week"
    onClick={[Function]}
  />
</div>
`;

exports[`MatchWeek renders without matches 1`] = `
<div
  className="MatchWeek"
>
  <h2>
    Week 
    1
  </h2>
  <p>
    TestSeason
  </p>
  <Matches />
  <SSButton
    disabled={false}
    label="Sim Week"
    onClick={[Function]}
  />
  <SSButton
    disabled={true}
    label="Advance Week"
    onClick={[Function]}
  />
</div>
`;

exports[`MatchWeek renders without season 1`] = `""`;
